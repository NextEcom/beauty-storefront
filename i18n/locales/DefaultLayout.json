{
  "en": {
    "change_language": "Change Language"
  },
  "ru": {
    "change_language": "Изменить язык"
  }
}
