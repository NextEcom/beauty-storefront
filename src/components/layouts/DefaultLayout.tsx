import { AvailableLocale } from "@/types";
import { Translate } from "@mui/icons-material";
import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Container from "@mui/material/Container";
import IconButton from "@mui/material/IconButton";
import Menu from "@mui/material/Menu";
import MenuItem from "@mui/material/MenuItem";
import Toolbar from "@mui/material/Toolbar";
import Typography from "@mui/material/Typography";
import { AppConfig } from "config/app";
import { useTranslations } from "next-intl";
import { useRouter } from "next/router";
import * as React from "react";

export const DefaultLayout = ({ children }: { children: React.ReactNode }) => {
  const { push, pathname } = useRouter();
  const t = useTranslations();

  const [anchorElLang, setAnchorElLang] = React.useState<null | HTMLElement>(
    null
  );

  const handleOpenLanguageMenu = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorElLang(event.currentTarget);
  };

  const handleCloseLanguageMenu = () => {
    setAnchorElLang(null);
  };

  const setLocale = (locale: AvailableLocale) => {
    return () => {
      handleCloseLanguageMenu();
      push(pathname, pathname, { locale });
    };
  };

  return (
    <>
      <AppBar position="sticky" elevation={1} color="inherit">
        <Container maxWidth="xl">
          <Toolbar disableGutters>
            <Typography variant="h6" noWrap component="div">
              LOGO
            </Typography>
            <Box flexGrow={1} />

            <Box sx={{ flexGrow: 0 }}>
              <IconButton
                aria-label={t("Common.change_language")}
                sx={{ mr: "25px" }}
                onClick={handleOpenLanguageMenu}
                color="inherit"
              >
                <Translate />
              </IconButton>
              <Menu
                anchorEl={anchorElLang}
                anchorOrigin={{
                  vertical: "top",
                  horizontal: "right",
                }}
                keepMounted
                transformOrigin={{
                  vertical: "top",
                  horizontal: "right",
                }}
                open={Boolean(anchorElLang)}
                sx={{ mt: `45px` }}
              >
                {AppConfig.AvailableLocales.map((locale) => (
                  <MenuItem key={locale} onClick={setLocale(locale)}>
                    {AppConfig.LocalesNames[locale]}
                  </MenuItem>
                ))}
              </Menu>
            </Box>
          </Toolbar>
        </Container>
      </AppBar>
      {children}
    </>
  );
};
