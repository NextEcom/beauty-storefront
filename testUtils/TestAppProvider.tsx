import { AvailableLocale } from "@/types";
import { AppConfig } from "config/app";
import I18nProvider from "i18n/I18nProvider";
import localesJson from "i18n/locales.json";
import nextRouterMock from "next-router-mock";
import { RouterContext } from "next/dist/shared/lib/router-context";
import { ReactNode } from "react";

export default function TestAppProvider({
  locale = AppConfig.defaultLocale,
  children,
}: {
  locale?: AvailableLocale;
  children: ReactNode;
}) {
  nextRouterMock.locale = locale;
  nextRouterMock.locales = AppConfig.AvailableLocales;
  nextRouterMock.defaultLocale = AppConfig.defaultLocale;
  return (
    <RouterContext.Provider value={nextRouterMock}>
      <I18nProvider locale={locale} messages={localesJson[locale]}>
        {children}
      </I18nProvider>
    </RouterContext.Provider>
  );
}
