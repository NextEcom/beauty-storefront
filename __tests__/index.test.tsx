import { render, screen } from "@testing-library/react";
import Index from "@/pages/index";
import DefaultLayout from "@/components/layouts/DefaultLayout";
import TestAppProvider from "testUtils/TestAppProvider";

describe("Home", () => {
  it("renders components", () => {
    render(
      <TestAppProvider>
        <DefaultLayout>
          <Index />
        </DefaultLayout>
      </TestAppProvider>
    );

    const heading = screen.getByRole("heading", {
      name: /welcome!/i,
    });

    const button = screen.getByRole("button", {
      name: /hi/i,
    });

    const changeLnBtn = screen.getByRole("button", {
      name: /change language/i,
    });

    expect(changeLnBtn).toBeInTheDocument();

    expect(heading).toBeInTheDocument();
    expect(button).toBeInTheDocument();
  });
});
