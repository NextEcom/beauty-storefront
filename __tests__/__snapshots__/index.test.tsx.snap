// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders components 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 MuiAppBar-root MuiAppBar-colorInherit MuiAppBar-positionSticky css-1dyihen-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular css-1cndkqf-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-8u39c-MuiTypography-root"
        >
          LOGO
        </div>
        <div
          class="MuiBox-root css-i9gxme"
        />
        <div
          class="MuiBox-root css-2uchni"
        >
          <button
            aria-label="Change Language"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-bveeel-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="TranslateIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m12.87 15.07-2.54-2.51.03-.03c1.74-1.94 2.98-4.17 3.71-6.53H17V4h-7V2H8v2H1v1.99h11.17C11.5 7.92 10.44 9.75 9 11.35 8.07 10.32 7.3 9.19 6.69 8h-2c.73 1.63 1.73 3.17 2.98 4.56l-5.09 5.02L4 19l5-5 3.11 3.11.76-2.04zM18.5 10h-2L12 22h2l1.12-3h4.75L21 22h2l-4.5-12zm-2.62 7 1.62-4.33L19.12 17h-3.24z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiBox-root css-6yat2x"
      data-testid="homepage-hero-banner"
    >
      <div
        class="MuiBox-root css-1l7c0cy"
      />
      <div
        class="css-2cww9k-MuiStack-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
        >
          Sub Heading
        </h4>
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
        >
          Heading
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Text
        </p>
        <a
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-21w08c-MuiButtonBase-root-MuiButton-root"
          href="https://example.com"
          tabindex="0"
        >
          Link Text
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="mui-1"
          id="mui-1-label"
        >
          First Name
          <span
            aria-hidden="true"
            class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="mui-1"
            name="firstName"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-173wfxe"
            >
              <span>
                First Name
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="mui-2"
          id="mui-2-label"
        >
          Last Name
          <span
            aria-hidden="true"
            class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="mui-2"
            name="lastName"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-173wfxe"
            >
              <span>
                Last Name
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="mui-3"
          id="mui-3-label"
        >
          Phone Number
          <span
            aria-hidden="true"
            class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="mui-3"
            name="phoneNumber"
            required=""
            type="tel"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-173wfxe"
            >
              <span>
                Phone Number
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="mui-4"
          id="mui-4-label"
        >
          Password
          <span
            aria-hidden="true"
            class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="mui-4"
            name="password"
            required=""
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-173wfxe"
            >
              <span>
                Password
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        Create Account
      </button>
    </div>
  </div>
</div>
`;
